scrapers_next/pa/committees.py

```diff
@@ -1,105 +1,603 @@
+            for member in members:
+                # In some cases, the member list is not a list of members, and
+                # is instead a reference to all members of the parent committee
+                if member == "All Commission Members (appointed and ex officio)":
+                    # Add all non-(vice)chair members from parent committee
+                    for parent_member in self.input.get("com").members:
+                        if parent_member.role in [
+                            "Member",
+                            "Public Member",
+                            "Ex-officio",
+                        ]:
+                            com.add_member(
+                                name=parent_member.name, role=parent_member.role
+                            )
+                else:
+                    # Add member based on the text
+                    memname, role = parse_name_and_role(member, None)
+                    com.add_member(name=memname, role=role)
+            yield check_enough_members(com)
+
+
+# Helper function for some of the Joint Committees that display
+# member names, titles, and roles in one block of text.
+# This function tries to extract the correct name and role.
+# A default role can be supplied when the role is already known
+def parse_name_and_role(name, role):
+    prefixes = {
+        "Judge ": "Public Member",
+        "Representative ": "Member",
+        "Senator ": "Member",
+        "Attorney ": "Public Member",
+        "Defense Attorney ": "Public Member",
+        "District Attorney ": "Public Member",
+        "Professor ": "Public Member",
+    }
+    # Remove any title prefixes in the name
+    for prefix, prefix_role in prefixes.items():
+        if name.startswith(prefix):
+            name = name[len(prefix) :]
+            # Override role if itbs not yet set
+            if role is None:
+                role = prefix_role
+
+    # Fall back on guessing that this is a public member if they have no role
+    # Normal members should always have the Senator/Representative prefix
+    if role is None:
+        role = "Public Member"
+
+    # Names may have suffixes related to that person's role in the parent committee
+    # These are treated like part of the name and need to be removed
+    # They must be listed individually, rather than removing all text after the comma,
+    # because some names contain commas like "<name>, Jr."
+    suffixes = [
+        ", Commission Chair (ex officio)",
+        ", Commission Chair",
+        ", Commission Vice Chair",
+        ", Commission Vice Chair (ex offico)",
+    ]
+    for suffix in suffixes:
+        if name.endswith(suffix):
+            name = name[: -len(suffix)]
+
+    return name, role
+
+
+class StateGovernmentComm(HtmlPage):
+    source = "http://jsg.legis.state.pa.us/about-us.cfm"
+
+    def process_page(self):
+        com = ScrapeCommittee(
+            name="State Government",
```

Comment:
Let's keep the full name of "State Government Commission" since it uses the word Commission rather than Committee.